examples/callback_w_args/callback_w_args.py

```diff
@@ -0,0 +1,15 @@
+import mido
+import queue
+import time
+
+rtmidi = mido.Backend('mido.backends.rtmidi')
+
+def myfunc(msg, q):
+    print(f'Got message: {msg}')
+    q.put(msg)
+
+q = queue.Queue()
+input_ = rtmidi.open_input('testing', virtual=True, callback_w_args=(myfunc, q))
```

Comment:
There's no need to implement anything in mido to achieve this. The standard way would be to use [`functools.partial`](https://docs.python.org/3.9/library/functools.html#functools.partial) to create a new callable that has some parameters set to something. (if I understand correctly what you want to achieve). Like this:

```py
input_ = rtmidi.open_input('testing', virtual=True, callback=partial(myfunc, q=q))
```

Reply:
Thank you! I did not know about "partial".

Reply:
Good question. Indeed you could use partial with Thread too. I guess args is there just for convenience as starting a Thread with some arguments is quite common.